uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestMailNotification.php

```diff
@@ -0,0 +1,391 @@
+  }
+
+  /**
+   * Returns the array of the option values for the LeaveRequest sickness_required_documents field.
+   *
+   * @return array
+   */
+  private function getSicknessRequiredDocuments() {
+    if (is_null($this->sicknessRequiredDocuments)) {
+      $result = civicrm_api3('OptionValue', 'get', [
+        'option_group_id' => 'hrleaveandabsences_leave_request_required_document',
+      ]);
+
+      $options = [];
+      foreach ($result['values'] as $requiredDocument) {
+        $options[$requiredDocument['value']] = $requiredDocument['label'];
+      }
+
+      $this->sicknessRequiredDocuments = $options;
+    }
+
+    return $this->sicknessRequiredDocuments;
+  }
+
+  /**
+   * Returns the array of the option values for the LeaveRequest from_date_type field.
+   *
+   * @return array
+   */
+  private function getLeaveRequestDayTypes() {
+    if (is_null($this->leaveRequestDayTypes)) {
+      $this->leaveRequestDayTypes = LeaveRequest::buildOptions('from_date_type');
+    }
+
+    return $this->leaveRequestDayTypes;
+  }
+
+  /**
+   * Gets the Comments associated with this LeaveRequest
+   *
+   * @param \CRM_HRLeaveAndAbsences_BAO_LeaveRequest $leaveRequest
+   *
+   * @return array
+   */
+  private function getLeaveComments(LeaveRequest $leaveRequest) {
+    $result = civicrm_api3('LeaveRequest', 'getComment', [
+      'leave_request_id' => $leaveRequest->id,
+      'api.Contact.get' => ['id' => '$value.contact_id', 'return' => ['display_name']]
+    ]);
+
+    array_walk($result['values'], function(&$item){
+      $item['commenter'] = $item['api.Contact.get']['values'][0]['display_name'];
+    });
+
+    return $result['values'];
+  }
+
+  /**
+   * Gets the Attachments associated with this LeaveRequest
+   *
+   * @param \CRM_HRLeaveAndAbsences_BAO_LeaveRequest $leaveRequest
+   *
+   * @return array
+   */
+  private function getAttachments(LeaveRequest $leaveRequest) {
+    $result = civicrm_api3('LeaveRequest', 'getAttachments', [
+      'leave_request_id' => $leaveRequest->id
+    ]);
+
+    return $result['values'];
+  }
+
+  /**
+   * Return URL for the leave request on SSP
+   *
+   * @return string
+   */
+  private function getLeaveRequestURL() {
+    return CRM_Utils_System::url('my-leave', [], true);
+  }
+
+  /**
+   * Return parameters to be used in the Email smarty template
+   *
+   * @param \CRM_HRLeaveAndAbsences_BAO_LeaveRequest $leaveRequest
+   *
+   * @return array
+   */
+  public function getTemplateParameters(LeaveRequest $leaveRequest) {
+    $templateParameters =  [
+      'leaveComments' => $this->getLeaveComments($leaveRequest),
+      'leaveFiles' => $this->getAttachments($leaveRequest),
+      'leaveRequestStatuses' => $this->getLeaveRequestStatuses(),
+      'fromDate' => $this->formatLeaveRequestDate($leaveRequest->from_date),
+      'toDate' => $this->formatLeaveRequestDate($leaveRequest->to_date),
+      'fromDateType' => $this->getLeaveRequestDayTypes()[$leaveRequest->from_date_type],
```

Comment:
Instead of using array dereferencing here, I think this will look better if you passe the values as a parameter, and then the method returns the label (the method name also has to be updated to say it returns a label. Something like this:

```php
$this->getLeaveRequestDayTypeLabel($leaveRequest->from_date_type);
```